### ISSUE TITLE
Select DependsOn: Set Value Not Emitting Change

### ISSUE THREAD

**Issue by 3owlcristian** [laravel/nova-issues#6380]
- Laravel Version:10.48.10
- Nova Version: 4.33.3 
- PHP Version:  8.1.28

Description:
I'm using a feature called 'dependOn' to automatically fill in the city, county, and state fields based on the postal code entered. The state and county fields are select fields with predefined options.

The functionality works correctly for text fields, but I'm experiencing issues with the select fields. Initially, when the select field has no value and I change the postal code, the correct value is selected based on the new postal code. However, after this initial selection, the value in the select field does not change anymore, even though I can see in the request that the value is correctly set for the field.

This issue only occurs on the update page. On the create page, the functionality works without any issues. I am also attaching video of this issue. https://www.awesomescreenshot.com/video/27543474?key=e26c0283592351a50ea820d7a7f4b33b
```
$request_data = request()->all();
$postal_info = null;
if (@$request_data['zip']) {
    $zip = str_pad($request_data['zip'], 6, '0', STR_PAD_LEFT);
    $postal_info = PostalInfo::where('zip', $zip)->first();
}
return [
  Text::make(__('Zip'), 'zip'),
  Text::make(__('City'), 'city')->dependsOn(
      'zip',
      function (Select $field, NovaRequest $request, FormData $formData) use ($postal_info) {
          if (@$postal_info) {
              $field->setValue($postal_info->city);
          }
      }
  ),
  Select::make(__('State'), 'state')->options(config('states'))->searchable()->hideFromIndex()->hideFromDetail()->nullable()->displayUsingLabels()->dependsOn(
      ['zip'],
      function (Select $field, NovaRequest $request, FormData $formData) use ($postal_info) {
          if (@$postal_info) {
              $field->setValue($postal_info->state);
          }
      }
  ),
  
  Select::make(__('County'), 'county_id')->options($counties_arr)->hideFromDetail()->searchable()->dependsOn(
      'zip',
      function (Select $field, NovaRequest $request, FormData $formData) use ($postal_info) {
          if (@$formData['zip']) {
              if (@$postal_info) {
                  $field->setValue($postal_info->county_id);
              }
  
          }
      }
  ),
];
```

**Comment by crynobone** [comment]
Don't do this the following.

```php
$request_data = request()->all();
$postal_info = null;
if (@$request_data['zip']) {
    $zip = str_pad($request_data['zip'], 6, '0', STR_PAD_LEFT);
    $postal_info = PostalInfo::where('zip', $zip)->first();
}
```

Only depends on value available via `FormData`>

**Comment by 3owlcristian** [comment]
@crynobone  I got your point but. With that getting same issue on update page.

The issue is happening with select field only on the update page! On create page it works without any issue!

**Comment by crynobone** [comment]
Again, this is not a supported behaviour.

**Comment by crynobone** [comment]
Unable to reproduce the issue, please provide full reproducing repository based on fresh installation as suggested in the bug report template (or you can refer to https://github.com/nova-issues for example)

**Comment by 3owlcristian** [comment]
@crynobone  please use this repo: https://github.com/3owlcristian/nova-dependson-issue

I have added some seeders also for testing postal codes.

Issue happing on **Contac Information** resource update page!

**Comment by crynobone** [comment]
Confirmed this is a bug but going to reserve this for a major release to avoid breaking existing application.

**Comment by 3owlcristian** [comment]
@crynobone  thank you for your response! Hope v5 will released soon!

**Comment by ecreeth** [comment]
@crynobone Why is this considered a breaking change if that is the expected behavior of the field.

> Confirmed this is a bug but going to reserve this for a major release to avoid breaking existing application.

**Comment by crynobone** [comment]
Changing behaviour on existing application is a breaking change. Would you be happy if a paid product out of nowhere cause your application to break to fix an expectation by someone else?

**Comment by ecreeth** [comment]
> Changing behaviour on existing application is a breaking change. Would you be happy if a paid product out of nowhere cause your application to break to fix an expectation by someone else?

I understand breaking changes. Just imagine that I have a production application running without knowing that a certain method isn't working as expected. When I was implementing a part of my application, I noticed it immediately. What would happen in the scenario that I didn't? The customer would had noticed it.

**Comment by salvisb** [comment]
@crynobone what about other paying customers who expect a feature to just work as it should? I am also curious as to why fixing a feature that is not working would break someone's application.

When might this be fixed? Thanks!

**Comment by sylouuu** [comment]
Hello here,

It seems the value is still updated when submitting the form and the issue is only on the UI feedback.

Hope that helps!

Bests

**Comment by ozanhazer** [comment]
I'm not sure if it's the same issue, but I'm having a similar one with the help text in a custom dependent form field:

```php
OrderContactField::make('Contact person')
    ->help('REQUIRED FOR SALES')
    ->dependsOn('tenant', function ($field, $request, $formData) {
        if ($formData->tenant) {
            $field->help('Some information');
        } else {
            $field->help('Other information');
        }
    });
```

It updates to "Some information" when a tenant is selected but the consequent updates are not reflected to the UI. The requests have the correct `helpText` in the responses.

I don't think it's caused by the custom field. FormField.vue:
```vue
<template>
  <DefaultField
    :field="field"
    :errors="errors"
    :show-help-text="showHelpText"
    :full-width-content="fullWidthContent"
  >
    <template #field>
    ...
    </template>
  </DefaultField>
</template>

<script>
import { DependentFormField, HandlesValidationErrors } from "laravel-nova"

export default {
  mixins: [DependentFormField, HandlesValidationErrors],
  ...
```

**Comment by crynobone** [comment]
@ozanhazer submit a proper bug report with reproducing code/repository.

**Comment by salvisb** [comment]
Hi @crynobone. Still looking forward to this fix. When could it be released? Thanks!

**Comment by crynobone** [comment]
Released with Laravel Nova 5.0.0

Feel free to open up a new issue if you're still experiencing this problem on the latest version.
